photutils/psf/epsf_stars.py

```diff
@@ -772,8 +772,20 @@ def _extract_stars(data, catalog, size=(11, 11), use_xy=True):
     if data.uncertainty is None:
         weights = np.ones_like(data.data)
     else:
+        # If NDData instance has been set to a custom uncertainty type with
+        # type ``weights`` then take the data as they are
         if data.uncertainty.uncertainty_type == 'weights':
             weights = np.asanyarray(data.uncertainty.array, dtype=np.float)
+        # Else, if astropy.nddata.StdDevUncertainty has been implemented for
+        # the data uncertainties we can take the inverse standard deviation as
+        # the weight. Before assigning the weights we must ignore zero
+        # variance data, however, setting the weights for these data points to
+        # zero by forcing their standard deviations to a large value.
+        elif data.uncertainty.uncertainty_type == 'std':
```

Comment:
Note that in the upcoming astropy release (3.1) it looks to me like there will be an easier way to "normalize" other uncertainties into std.  We'll want to do that for this option I'd think, but no point in doing that until the astropy release is available for testing.

Reply:
So is this a "accept this and we'll revisit it later" or a "hold off on this until after the release is available for testing"? I think it would be good in the future to expand this quite a fair amount to accept various kinds of uncertainty (perhaps parsing them through some convenience function to produce an internally accepted 'standard' type), and it sounds like this will help with that down the line.

Reply:
Astropy 3.1 is out, so I've assigned @mwcraig to this issue in case he can help with this :)

Reply:
Right now what you could is something like this:

```python
uncertainty_conversion_to_std = {
     'std': lambda x: x,  # do nothing
     'var': np.sqrt,  # square root of the variance is the standard dev
     'ivar': lambda x: 1 / np.sqrt(x)   # invert and square root to get standard dev
}

uncert = data.uncertainty.array.copy()
std_dev = uncertainty_conversion_to_std[data.uncertainty.type](uncert)
```

(note: I haven't tested this code but hopefully it conveys the idea).